### PR TITLE
TryValidateObjectRecursive was improved to validate classes with List…

### PR DESCRIPTION
**Pull request by borismod**
TryValidateObjectRecursive was improved to validate classes with List of Dictionaries

### CODE REVIEW

File: DataAnnotationsValidator/DataAnnotationsValidator.Tests/DataAnnotationsValidatorTests.cs [hunk 1]
```diff
@@ -157,6 +157,29 @@ public void TryValidateObject_modifies_membernames_for_nested_properties()
 			Assert.AreEqual(1, validationResults.Count);
 			Assert.AreEqual("Child PropertyA is required", validationResults[0].ErrorMessage);
 			Assert.AreEqual("Child.PropertyA", validationResults[0].MemberNames.First());
-		}
+		}
+
+        [Test]
```
Comment (reustmd) [inline DataAnnotationsValidator/DataAnnotationsValidator.Tests/DataAnnotationsValidatorTests.cs]: looks like indentation is off?

File: DataAnnotationsValidator/DataAnnotationsValidator.Tests/DataAnnotationsValidatorTests.cs [hunk 2]
```diff
@@ -157,6 +157,29 @@ public void TryValidateObject_modifies_membernames_for_nested_properties()
...
+            {
+                Objects =
+                    new List<Dictionary<string, Child>>{
+                    new Dictionary<string, Child>
+                {
+                    { "key", new Child
+                    {
+                        PropertyA = 1,
+                    PropertyB = 2
+                    } } } }
```
Comment (reustmd) [inline DataAnnotationsValidator/DataAnnotationsValidator.Tests/DataAnnotationsValidatorTests.cs]: can you clean up the construction of this class a bit? all the nesting leaves this a bit gnarly.

File: DataAnnotationsValidator/DataAnnotationsValidator/DataAnnotationsValidator.cs [hunk 3]
```diff
@@ -17,7 +17,7 @@ public bool TryValidateObjectRecursive<T>(T obj, List<ValidationResult> results)
 		{
 			bool result = TryValidateObject(obj, results);
 
-			var properties = obj.GetType().GetProperties().Where(prop => prop.CanRead && !prop.GetCustomAttributes(typeof(SkipRecursiveValidation), false).Any()).ToList();
+            var properties = obj.GetType().GetProperties().Where(prop => prop.CanRead && !prop.GetCustomAttributes(typeof(SkipRecursiveValidation), false).Any() && prop.GetIndexParameters().Length == 0).ToList();
```
Comment (reustmd) [inline DataAnnotationsValidator/DataAnnotationsValidator/DataAnnotationsValidator.cs]: 🚑 indentation

**Comment by reustmd** [comment]
Logically the change is g2g, but please cleanup the formatting a bit. Sorry for the nit.

File: DataAnnotationsValidator/DataAnnotationsValidator.Tests/DataAnnotationsValidatorTests.cs (hunk 1 above)
Comment (borismod) [inline DataAnnotationsValidator/DataAnnotationsValidator.Tests/DataAnnotationsValidatorTests.cs]: Looks like spaces vs. tabs. Tried to fix it

File: DataAnnotationsValidator/DataAnnotationsValidator/DataAnnotationsValidator.cs (hunk 3 above)
Comment (borismod) [inline DataAnnotationsValidator/DataAnnotationsValidator/DataAnnotationsValidator.cs]: Again tabs vs. spaces. Fixed as well.

**Comment by reustmd** [comment]
thanks @borismod

**Pull request merged**